images/server/install-packages.sh

```diff
@@ -10,20 +10,28 @@ need_curl() {
     dnf install --setopt=install_weak_deps=False -y /usr/bin/curl
 }
 
-get_custom_repo() {
-    url="$1"
-    fname="$(basename "$url")"
-    dest="/etc/yum.repos.d/${fname}"
-    while [ -e "$dest" ]; do
-        u="$(echo "${url}" | sha256sum | head -c12)"
-        if [ -z "$u" ]; then
-            echo "failed to uniquify repo file name"
-            exit 2
+get_custom_repos() {
+    if [[ -n $1 ]]; then
+        urls=$1
+        need_curl
+        if [[ $(wc -w <<< "$1") -eq 1 || $ceph_from_custom -ne 1 ]]; then
+            get_distro_ceph_repo
+            urls=${1%% *}
```

Comment:
This won't work if, for some reason, the string starts with a space. Why not use bash arrays ?

```bash
urls=(${1})
...
if [[ ${#urls[@]} -eq 1 || $ceph_from_custome -ne 1 ]]; then
    ...
    urls=(${urls[0]})
fi
for url in "${urls[@]}"; do
...
```